js_modules/dagit/packages/core/src/workspace/WorkspaceContext.tsx

```diff
@@ -273,7 +284,32 @@ const useVisibleRepos = (
     [setHiddenKeys],
   );
 
-  const visibleOptions = React.useMemo(() => {
+  const setVisible = React.useCallback(
+    (repoAddresses: RepoAddress[]) => {
+      const keysToShow = new Set(
+        repoAddresses.map((repoAddress) => `${repoAddress.name}:${repoAddress.location}`),
+      );
+      setHiddenKeys((current) => {
+        return current?.filter((key) => !keysToShow.has(key));
+      });
+    },
+    [setHiddenKeys],
+  );
+
+  const setHidden = React.useCallback(
+    (repoAddresses: RepoAddress[]) => {
+      const keysToHide = new Set(
+        repoAddresses.map((repoAddress) => `${repoAddress.name}:${repoAddress.location}`),
+      );
+      setHiddenKeys((current) => {
+        const updatedSet = new Set([...(current || []), ...keysToHide]);
+        return Array.from(updatedSet);
+      });
+    },
+    [setHiddenKeys],
+  );
```

Comment:
Added these to `useVisibleRepos` because they allow better control of multiple repos vs. `toggleVisible`.